volmaster/daemon.go

```diff
@@ -129,16 +134,32 @@ func (d daemonConfig) handleUnmount(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func (d daemonConfig) handleMount(w http.ResponseWriter, r *http.Request) {
+func (d daemonConfig) handleMountWithTTLFlag(w http.ResponseWriter, r *http.Request, exist client.PrevExistType) error {
 	req, err := unmarshalUseConfig(r)
 	if err != nil {
-		httpError(w, "Unmarshalling request", err)
-		return
+		return err
 	}
 
 	req.Reason = "Mount"
 
-	if err := d.config.PublishUse(req); err != nil {
+	fmt.Println(req)
+
+	if err := d.config.PublishUseWithTTL(req, time.Duration(d.mountTTL)*time.Second, exist); err != nil {
```

Comment:
This is just a thought that I got while reviewing this PR. I thought to bring it up in case you already didn't consider and rejected it due to other implications that I might have missed.

I think using the `d.mountTTL` (passed to volmaster as a flag) creates a dependency between ttl values set for volplugin and volmaster at their startup, for instance if I got it right we need to set volplugin ttl to be smaller (or equal to) that volmaster's ttl for the current approach to mostly work. I think part of this comes from that fact that we let volmaster update the runtime state for a mount created by volplugin in etcd.

Alternatively, would it be simpler if we let volplugin update this state in etcd itself? So volmaster stores the configuration state of volume and distributes it to volplugin. While volplugin's update the runtime mount state in etcd. In an event where two volplugin instances try to create same state I think we can let etcd fail one of the writes.